src/dos/program_more.cpp

```diff
@@ -60,15 +61,79 @@ bool MORE::ParseCommandLine(MoreOutputFiles &output)
 	cmd->FillVector(params);
 
 	// Check if specified tabulation size
-	if (!params.empty()) {
-		const auto &param = params[0];
-		if ((starts_with("/t", param) || starts_with("/T", param)) &&
-		    (param.length() == 3) && (param.back() >= '1') &&
-		    (param.back() <= '9')) {
-			// FreeDOS extension - custom TAB size
-			output.SetTabSize(static_cast<uint8_t>(param.back() - '0'));
+	while (!params.empty()) {
+		auto is_option_extended = [](const std::string& param) {
+			return param == "/e" || param == "/E";
+		};
+		auto is_option_clear = [](const std::string& param) {
+			return param == "/c" || param == "/C";
+		};
+		auto is_option_expand_form_feed = [](const std::string& param) {
+			return param == "/p" || param == "/P";
+		};
+		auto is_option_squish = [](const std::string& param) {
+			return param == "/s" || param == "/S";
+		};
+		auto is_option_start_line = [](const std::string& param) {
+			if ((param.length() < 2) || (param[0] != '+')) {
+				return false;
+			}
+			for (size_t i = 1; i < param.length(); ++i) {
+				if (!std::isdigit(param[i])) {
+					return false;
+				}
+			}
+			return true;
+		};
+		auto is_option_tab_size = [](const std::string& param) {
+			return (param.length() == 3) &&
+			       (starts_with("/t", param) || starts_with("/T", param)) &&
+			       (param.back() >= '1') && (param.back() <= '9');
+		};
+
+		const auto& param = params[0];
+		if (is_option_extended(param)) {
+			output.SetOptionExtendedMode(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_clear(param)) {
+			output.SetOptionClear(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_expand_form_feed(param)) {
+			output.SetOptionExpandFormFeed(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_squish(param)) {
+			output.SetOptionSquish(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_start_line(param)) {
+			int32_t line_num = 0;
+			try {
+				line_num = std::stoi(param);
+			} catch (...) {
+				result_errorcode = DOSERR_FUNCTION_NUMBER_INVALID;
+				WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"),
+				         param.c_str());
+				return false;
+			}
+			output.SetOptionStartLine(static_cast<uint32_t>(line_num));
 			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_tab_size(param)) {
+			// FreeDOS / modern Windows extension - custom TAB size
+			const auto size = static_cast<uint8_t>(param.back() - '0');
+			output.SetOptionTabSize(size);
```

Comment:
Is it possible to enter an ASCII char prior to `'0'`? This would produce a negative value when `'0'` is subtracted and then wraps to a large ~250 type value in the cast-to-`uint8_t`?

Reply:
The `is_option_tab_size` checks whether the value is valid. And 0 is not allowed either, only 1-9.